### ISSUE TITLE
Artist Download Error

### ISSUE THREAD

**Issue by Zibbp** [MinisculeGirraffe/tdl#12]
Hi, 

I'm trying to download an entire artist, not sure if it's officially supported yet, but I thought I would give it a try. It worked for a few until it got to one and threw an error. The command I'm using is `./tdl get https://listen.tidal.com/artist/19137`.

Error output:
```
Access Token Valid
Getting Artist Albums   thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Error at line 1 column 12942', src/api/media.rs:69:53
stack backtrace:
   0:     0x55a6cd4195e1 - <unknown>
   1:     0x55a6cd1e374c - <unknown>
   2:     0x55a6cd3f5874 - <unknown>
   3:     0x55a6cd41a5b6 - <unknown>
   4:     0x55a6cd41b646 - <unknown>
   5:     0x55a6cd41b0e2 - <unknown>
   6:     0x55a6cd41b056 - <unknown>
   7:     0x55a6cd41b012 - <unknown>
   8:     0x55a6cd0d6be2 - <unknown>
   9:     0x55a6cd0d6e62 - <unknown>
  10:     0x55a6cd45f998 - <unknown>
  11:     0x55a6cd48a881 - <unknown>
  12:     0x55a6cd4a95f7 - <unknown>
  13:     0x55a6cd4abece - <unknown>
  14:     0x55a6cd498e45 - <unknown>
  15:     0x55a6cd49dffb - <unknown>
  16:     0x55a6cd41c0c5 - <unknown>
  17:     0x7fa5032fa609 - start_thread
  18:     0x7fa5030c8163 - clone
  19:                0x0 - <unknown>
```

Full log with `RUST_LOG=trace` enabled.

```
[2022-07-28T19:55:00Z DEBUG tdl::login] Attempting to validate access token
[2022-07-28T19:55:00Z DEBUG reqwest::connect] starting new connection: https://api.tidalhifi.com/
[2022-07-28T19:55:00Z TRACE mio::poll] registering event source with poller: token=Token(1), interests=READABLE | WRITABLE
[2022-07-28T19:55:00Z TRACE want] signal: Want
[2022-07-28T19:55:00Z TRACE want] signal found waiting giver, notifying
[2022-07-28T19:55:00Z TRACE want] poll_want: taker wants!
[2022-07-28T19:55:00Z TRACE want] signal: Want
[2022-07-28T19:55:00Z TRACE want] signal: Want
[2022-07-28T19:55:00Z DEBUG reqwest::async_impl::client] response '200 OK' for https://api.tidalhifi.com/v1/sessions
Access Token Valid
[2022-07-28T19:55:00Z TRACE want] signal: Want
Getting Artist Albums                                                                                                                                                                      [2022-07-28T19:55:00Z DEBUG reqwest::connect] starting new connection: https://api.tidal.com/
[2022-07-28T19:55:00Z DEBUG reqwest::connect] starting new connection: https://api.tidal.com/
[2022-07-28T19:55:00Z TRACE mio::poll] registering event source with poller: token=Token(2), interests=READABLE | WRITABLE
[2022-07-28T19:55:00Z TRACE mio::poll] registering event source with poller: token=Token(3), interests=READABLE | WRITABLE
[2022-07-28T19:55:00Z TRACE want] signal: Want
[2022-07-28T19:55:00Z TRACE want] signal found waiting giver, notifying
[2022-07-28T19:55:00Z TRACE want] poll_want: taker wants!
[2022-07-28T19:55:00Z TRACE want] signal: Want
[2022-07-28T19:55:00Z TRACE want] signal found waiting giver, notifying
[2022-07-28T19:55:00Z TRACE want] poll_want: taker wants!
[2022-07-28T19:55:01Z DEBUG reqwest::async_impl::client] response '200 OK' for https://api.tidal.com/v1/artists/19137/albums?limit=50&offset=0&countryCode=US
[2022-07-28T19:55:01Z TRACE want] signal: Want
[2022-07-28T19:55:01Z TRACE want] signal found waiting giver, notifying
[2022-07-28T19:55:01Z TRACE want] signal: Want
[2022-07-28T19:55:01Z TRACE want] poll_want: taker wants!
[2022-07-28T19:55:01Z DEBUG tdl::api] {"limit":50,"offset":0,"totalNumberOfItems":18,"items":[{"id":160076435,"title":"POST HUMAN: SURVIVAL HORROR","duration":1932,"streamReady":true,"streamStartDate":"2020-10-30T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":9,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2020-10-30","copyright":"(P) 2020 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/160076435","cover":"a34e046e-5ca9-418f-87d5-a536966ce596","vibrantColor":"#d999af","videoCover":null,"explicit":true,"upc":"886448602200","popularity":45,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":164297268,"title":"POST HUMAN: SURVIVAL HORROR","duration":1931,"streamReady":true,"streamStartDate":"2020-12-03T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":9,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2020-10-30","copyright":"(P) 2020 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/164297268","cover":"5f7cfe77-13ac-4d34-8cf7-b7a0778b30a0","vibrantColor":"#d999af","videoCover":null,"explicit":false,"upc":"886448910428","popularity":5,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":160515739,"title":"POST HUMAN: SURVIVAL HORROR","duration":1932,"streamReady":true,"streamStartDate":"2020-10-30T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":9,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2020-10-30","copyright":"(P) 2020 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/160515739","cover":"4d4fbcf1-af50-485f-8422-d43265c370ae","vibrantColor":"#d999af","videoCover":null,"explicit":true,"upc":"886448846390","popularity":18,"audioQuality":"LOW","audioModes":["SONY_360RA"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":165492381,"title":"Live at the Royal Albert Hall","duration":5012,"streamReady":true,"streamStartDate":"2020-12-18T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":16,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2020-10-28","copyright":"(P) 2020 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/165492381","cover":"46922d26-c7e9-442a-a84d-a3f95236e002","vibrantColor":"#9e4243","videoCover":null,"explicit":false,"upc":"886448959434","popularity":14,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":176307422,"title":"Live at the Royal Albert Hall","duration":4982,"streamReady":true,"streamStartDate":"2021-03-12T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":16,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2020-10-28","copyright":"(P) 2021 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/176307422","cover":"66cc908f-0df1-45bb-9ace-7402e0d938e4","vibrantColor":"#9e4243","videoCover":null,"explicit":false,"upc":"886448988700","popularity":3,"audioQuality":"LOW","audioModes":["SONY_360RA"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":126396297,"title":"Music to listen to~dance to~blaze to~pray to~feed to~sleep to~talk to~grind to~trip to~breathe to~help to~hurt to~scroll to~roll to~love to~hate to~learn Too~plot to~play to~be to~feel to~breed to~sweat to~dream to~hide to~live to~die to~GO TO","duration":4533,"streamReady":true,"streamStartDate":"2019-12-27T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":8,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2019-12-27","copyright":"(P) 2019 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/126396297","cover":"659f6902-864e-487d-a7c9-7c2488501454","vibrantColor":"#984a41","videoCover":null,"explicit":true,"upc":"886448213192","popularity":19,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":102519387,"title":"amo","duration":3119,"streamReady":true,"streamStartDate":"2019-01-25T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":13,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2019-01-25","copyright":"(P) 2019 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/102519387","cover":"743df668-55eb-445c-abf7-f24c67049f30","vibrantColor":"#c4af24","videoCover":null,"explicit":true,"upc":"886447279588","popularity":42,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":103198350,"title":"amo","duration":3119,"streamReady":true,"streamStartDate":"2020-09-16T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":13,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2019-01-25","copyright":"(P) 2019 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/103198350","cover":"86076c01-a7c3-43aa-8e70-57b52b7246f2","vibrantColor":"#c4af24","videoCover":null,"explicit":false,"upc":"886447526255","popularity":4,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":121133400,"title":"2004 - 2013","duration":3978,"streamReady":true,"streamStartDate":"2017-11-24T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":15,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2017-11-24","copyright":"Visible Noise / Sony Music Entertainment UK Ltd., under exclusive license to Epitaph","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/121133400","cover":"df3721f1-6a47-49f5-8aed-01e34d0b0609","vibrantColor":"#FFFFFF","videoCover":null,"explicit":true,"upc":"0045778756063","popularity":7,"audioQuality":"LOSSLESS","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":50793895,"title":"That's The Spirit","duration":2703,"streamReady":true,"streamStartDate":"2015-09-11T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":11,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2015-09-11","copyright":"(P) 2015 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/50793895","cover":"bc9e6d77-e0fd-4c0d-a6f9-ff33fa07184b","vibrantColor":"#FFFFFF","videoCover":null,"explicit":true,"upc":"886445386097","popularity":58,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":152008773,"title":"That's The Spirit","duration":2706,"streamReady":true,"streamStartDate":"2020-08-17T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":11,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2015-09-11","copyright":"(P) 2015 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/152008773","cover":"cf139ba0-4ea9-4ddb-90d4-e575a63042be","vibrantColor":"#FFFFFF","videoCover":null,"explicit":false,"upc":"886445388244","popularity":3,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":211802103,"title":"Sempiternal (Deluxe)","duration":3967,"streamReady":true,"streamStartDate":"2021-04-01T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":13,"numberOfVideos":1,"numberOfVolumes":1,"releaseDate":"2013-04-01","copyright":"(P) 2013 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/211802103","cover":"cb3faefe-81a2-4bad-a5c3-9d06074b17b5","vibrantColor":"#ac8025","videoCover":null,"explicit":true,"upc":"886443859111","popularity":4,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":19291979,"title":"Sempiternal (Expanded Edition)","duration":3218,"streamReady":true,"streamStartDate":"2021-04-01T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":13,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2013-04-01","copyright":"(P) 2013 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/19291979","cover":"f49be36d-4032-4c61-b1aa-85b9b0833e18","vibrantColor":"#ae8637","videoCover":null,"explicit":true,"upc":"886443690707","popularity":56,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":19440652,"title":"Sempiternal","duration":2736,"streamReady":true,"streamStartDate":"2021-04-01T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":11,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2013-04-01","copyright":"(P) 2013 Sony Music Entertainment UK Limited","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/19440652","cover":"feaef112-ec93-4848-8050-928be7e4de80","vibrantColor":"#ac8025","videoCover":null,"explicit":true,"upc":"886443690714","popularity":14,"audioQuality":"HI_RES","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":121131541,"title":"There Is A Hell Believe Me I've Seen It. There Is A Heaven Let's Keep It A Secret","duration":3177,"streamReady":true,"streamStartDate":"2010-10-05T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":12,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2010-10-05","copyright":"Visible Noise Ltd. Under Exclusive License To Epitaph.","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/121131541","cover":"a1010f2e-5a36-40c0-a8dd-840bf606d12e","vibrantColor":"#d6bf94","videoCover":null,"explicit":true,"upc":"0045778706563","popularity":24,"audioQuality":"LOSSLESS","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":121131302,"title":"Suicide Season (Deluxe Edition)","duration":6317,"streamReady":true,"streamStartDate":"2010-04-13T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":24,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2008-01-01","copyright":"Visible Noise Ltd. Under Exclusive License To Epitaph","type":"ALBUM","version":"Deluxe Edition","url":"http://www.tidal.com/album/121131302","cover":"0a6ff36a-1eef-448c-b02f-9dee2c3c520d","vibrantColor":"#d5e2a5","videoCover":null,"explicit":true,"upc":"0045778705160","popularity":18,"audioQuality":"LOSSLESS","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":121129402,"title":"Suicide Season","duration":2535,"streamReady":true,"streamStartDate":"2008-09-30T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":10,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2008-01-01","copyright":"Visible Noise Limited Under Exclusive License To Epitaph","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/121129402","cover":"a033df3b-b4cb-45bd-83c4-69edc18588c4","vibrantColor":"#dd7467","videoCover":null,"explicit":true,"upc":"0045778700660","popularity":24,"audioQuality":"LOSSLESS","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]},{"id":121132771,"title":"Count Your Blessings","duration":2176,"streamReady":true,"streamStartDate":"2013-11-26T00:00:00.000+0000","allowStreaming":true,"premiumStreamingOnly":false,"numberOfTracks":10,"numberOfVideos":0,"numberOfVolumes":1,"releaseDate":"2006-10-30","copyright":"Visible Noise Limited, under exclusive license to Epitaph","type":"ALBUM","version":null,"url":"http://www.tidal.com/album/121132771","cover":"383c039c-5a3a-4a3a-9dfd-d6a639572f80","vibrantColor":"#9adde9","videoCover":null,"explicit":true,"upc":"0045778727964","popularity":18,"audioQuality":"LOSSLESS","audioModes":["STEREO"],"artist":{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"},"artists":[{"id":19137,"name":"Bring Me The Horizon","type":"MAIN","picture":"c34fb076-fc13-4b96-a89a-0e89750c7377"}]}]}
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Error at line 1 column 2320', src/api/media.rs:69:53
stack backtrace:
[2022-07-28T19:55:01Z TRACE want] signal: Want
   0:     0x56335c6305e1 - <unknown>
   1:     0x56335c3fa74c - <unknown>
   2:     0x56335c60c874 - <unknown>
   3:     0x56335c6315b6 - <unknown>
   4:     0x56335c632646 - <unknown>
   5:     0x56335c6320e2 - <unknown>
   6:     0x56335c632056 - <unknown>
   7:     0x56335c632012 - <unknown>
   8:     0x56335c2edbe2 - <unknown>
   9:     0x56335c2ede62 - <unknown>
  10:     0x56335c676998 - <unknown>
  11:     0x56335c6a1881 - <unknown>
  12:     0x56335c6c05f7 - <unknown>
  13:     0x56335c6c2ece - <unknown>
  14:     0x56335c6afe45 - <unknown>
  15:     0x56335c6b4ffb - <unknown>
  16:     0x56335c6330c5 - <unknown>
  17:     0x7fd43a8c7609 - start_thread
  18:     0x7fd43a695163 - clone
  19:                0x0 - <unknown>
[2022-07-28T19:55:01Z TRACE mio::poll] deregistering event source from poller
[2022-07-28T19:55:01Z TRACE mio::poll] deregistering event source from poller
[2022-07-28T19:55:01Z TRACE want] signal: Closed
[2022-07-28T19:55:01Z TRACE want] signal: Closed
[2022-07-28T19:55:01Z TRACE mio::poll] deregistering event source from poller
[2022-07-28T19:55:01Z TRACE want] signal: Closed
```

**Comment by MinisculeGirraffe** [comment]
Seems to be a value returned from the API that mismatched with the structs causing a parsing error. Should be able to write a patch for it pretty easily with the log. 
 
 Appreciate you gathering it.

**Comment by MinisculeGirraffe** [comment]
Ah, The Audiomode was a value i'd never seen from the API before.  

`"audioModes":["SONY_360RA"]`

I can add it as an enum value in the morning and i'll author a new release that includes the fix. Want to also do some additional testing to make sure that the files it downloads actually work.

**Comment by MinisculeGirraffe** [comment]
Just pushed the fix and created v0.2.3. The CI will take a bit to compile the binaries, but they'll be available at the link below shortly.

https://github.com/MinisculeGirraffe/tdl/releases/tag/v0.2.3
